src/query/models/query_context.go

```diff
@@ -42,7 +43,15 @@ type QueryContext struct {
 type QueryContextOptions struct {
 	// LimitMaxTimeseries limits the number of time series returned by each
 	// storage node.
-	LimitMaxTimeseries int
+	LimitMaxTimeseries      int
```

Comment:
Would it make sense to put this under `RestrictFetchTimeseries` opts also?

Reply:
Hm I should probably just rename `RestrictFetchTimeseries`.  I need a struct that is `nil` able to be able to tell whether you want to restrict to a specific namespace.

Reply:
Ok changed it to `RestrictFetchType` to make distinct difference to just generic restriction options.